Add support for Laravel 11 and 12

Meeshalk opened the pull request:
* Updated supported Illuminate packages
* Updated dev dependencies based on required dependencies
* Updated phpunit configuration using `--migrate-configuration` flag

Meeshalk committed:
fix: added additional supported versions for illuminate db and support

Meeshalk committed:
fix: updated req-dev packages to support new req packages

Meeshalk committed:
fix: migrated phpunit configuration

Meeshalk committed:
Merge pull request #1 from Meeshalk/Chore/Update-dependencies-for-laravel-11

Chore/update dependencies for laravel 11 & 12

Meeshalk commented:
Hi @santigarcor!

I am using this package in one of my project and while migrating it to L11. Came across an issue, that this package doesn't support laravel 11 or 12. 

Which was just because of the missing supported version in the composer.json file.

While updating phpunit was giving deprecation notice to its config file, based on their docs, I used this `--migrate-configuration` flag to update the configuration xml file.

All the tests are running fine now and also has support for L11 & L12.

santigarcor commented:
I think you need to update the matrix because currently the test matrix is failing

Meeshalk committed:
fix: updated php version matrix to include new versions

Meeshalk committed:
fix: updated github actions steps versions

Meeshalk committed:
fix: verbose flag is not longer supported by phpunit

Meeshalk committed:
fix: PHP 7.2 is no longer supported by laravel/legacy-factories

```
Problem 1
    - Root composer.json requires laravel/legacy-factories ^1.3 -> satisfiable by laravel/legacy-factories[v1.3.0, ..., 1.x-dev].
    - laravel/legacy-factories[v1.3.0, ..., 1.x-dev] require php ^7.3|^8.0 -> your php version (7.2.34) does not satisfy that requirement.
```

Meeshalk committed:
feat: adding supported PHP versions in composer.json

Meeshalk committed:
fix: removing 8.4 as 8.3 is well supported by Laravel 12

Meeshalk committed:
fix: updated require and dev dependencies to fix --prefer-lowest issues

* for PHP8.0 mockery 1.3.3 is required (dependency of orchestra/testbench)
* phpunit below version 9 has direct conflict with PHP8 and beyond
* orchestra/testbench lowest version for PHP8 and beyond is v6.8
* and orchestra/testbench v6 does not support Laravel 6 & 7

Meeshalk committed:
fix: PHP 7.3 is no longer needed

Meeshalk committed:
fix: PHP7 cannot be supported along with PHP8 with --prefer-lowest flag

This means Laravel 8 is no longer supported, so removed PHP7 and along with Laravel 8 support

Meeshalk committed:
fix: try to test with PHP7.4

Meeshalk committed:
fix: redo PHP7 removal

Meeshalk committed:
Merge pull request #2 from Meeshalk/Chore/update-version-support-and-github-actions

Chore/update version support and GitHub actions

* Removes support of php7
* And with that removed support for Laravel 6, 7 & 8

Meeshalk commented:
I tried to keep as much supported Laravel versions but PHP8 is just a big breaking change.
If we remove `--prefer-lowest` tests then we can support from L7 to L12. Otherwise, L9 to L12.

Current Version:
![image](https://github.com/user-attachments/assets/97bd29b8-b54b-4254-bfe0-6918f71e1c56)

With PHP7 and L7 & 8 supported:
![image](https://github.com/user-attachments/assets/1b70f4a7-f222-4880-a340-3ea7a7fbacae)

Anyways, Let me know if you need anything else to update.

Meeshalk committed:
fix: try with minimun supported version

Meeshalk committed:
fix: add phpunit matrix and exclude not supported versions

Meeshalk committed:
fix: add PHP7.3 & 7.4 exlusions

Meeshalk committed:
fix: PHP 8.0 is not supported by PHPUnit

Meeshalk committed:
fix: PHPUnit 8 & 9 does sot support PHP 8.1, 8.2 & 8.3

Meeshalk committed:
fix: try more on php 8.1

Meeshalk committed:
fix: PHP 8.1 only seems to work with PHPUnit 10

Meeshalk committed:
Merge pull request #3 from Meeshalk/Fix/use-exclude-for-unsuported-versions

Fix/use exclude for unsuported versions

Meeshalk commented:
@santigarcor 
I have added support for L6, 7 and 8 by using exclude on test matrix.
Please do check, I am using this package on one of my projects and it would help a lot if you could test and merge. 
Thanks 🙂

santigarcor commented:
@Meeshalk If you're adding support for Laaravel 11 and 12 jjust add to the matrix the version supported by those versions which is PHP >= 8.2, and "Drop" support for the others.

Meeshalk committed:
fix: PHP 8.3 tests removed

Meeshalk committed:
Merge pull request #4 from Meeshalk/Fix/remove-unnecessary-php-version

fix: PHP 8.3 tests removed

santigarcor commented:
You removed the wrong ones, I meant to only add support for php greater or equal than 8.2. Because this update adds the support for Laravel 11 and 12.

Meeshalk commented:
@santigarcor 
you mean from `matrix.exclude` in tests.yml?

I have added new phpunit versions which doesn't support php 7.*. So, I don't understand what you mean. 

Please review composer.json and let me know whats wrong?

isclopezm commented:
How can I help you @santigarcor or @Meeshalk to have this PR merged in order to release the new version so we can use this package in a Laravel 12 project.

santigarcor commented:
This update should only support Laravel 11 and 12.
